tgui/packages/tgui/interfaces/ChemDispenser.js

```diff
@@ -0,0 +1,439 @@
+  },
+};
+
+export const ChemDispenser = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    beakerContents,
+  } = data;
+  return (
+    <Window
+      width={570}
+      height={705}
+      theme="ntos">
+      <Window.Content scrollable>
+        <Box>
+          <ReagentDispenser />
+          <Beaker />
+          {!!beakerContents.length && (
+            <BeakerContentsGraph />
+          )}
+          <ChemGroups />
+        </Box>
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const ReagentDispenser = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    beakerName,
+    currentBeakerName,
+    maximumBeakerVolume,
+    beakerTotalVolume,
+  } = data;
+  const [addAmount, setAddAmount] = useLocalState(context, 'addAmount', 20);
+  const [iconToggle, setIconToggle] = useSharedState(context, 'iconToggle', false);
+  const [hoverOverId, setHoverOverId] = useLocalState(context, 'hoverOver', "");
+
+  const dispensableReagents = data.dispensableReagents || [];
+
+  return (
+    <Section
+      fontSize="12px"
+      title={(
+        <Fragment>
+          Dispense
+          <Box
+            as="span"
+            ml={18}>
+            Icons:
+            <Button
+              width={2}
+              textAlign="center"
+              backgroundColor="rgba(0, 0, 0, 0)"
+              textColor={iconToggle ? "rgba(255, 255, 255, 0.5)" : "rgba(255, 255, 255, 1)"}
+              onClick={() => setIconToggle(false)}>
+              <Icon mr={1} name={"circle"} />
+            </Button>
+            <Button
+              width={2}
+              backgroundColor="rgba(0, 0, 0, 0)"
+              textColor={iconToggle ? "rgba(255, 255, 255, 1)" : "rgba(255, 255, 255, 0.5)"}
+              onClick={() => setIconToggle(true)}>
+              <Icon name={"tint"} />
+            </Button>
+          </Box>
+        </Fragment>
+      )}
+      buttons={(
+        <Box>
+          {"Dispense Amount: "}
+          <NumberInput
+            value={addAmount}
+            format={value => value + "u"}
+            width={4}
+            minValue={1}
+            maxValue={100}
+            onDrag={(e, value) => setAddAmount(value)} />
+        </Box>
+      )}>
+      <Section fitted backgroundColor="rgba(0,0,0,0)">
+        {(!maximumBeakerVolume || maximumBeakerVolume
+        === beakerTotalVolume) && (
+          <Modal
+            className="chem-dispenser__labels"
+            fontSize="20px"
+            mr={2}
+            p={3}>
+            <Box>
+              {!maximumBeakerVolume &&(
+                "No " + beakerName + " Inserted"
+              ) || currentBeakerName + " Full"}
+            </Box>
+          </Modal>
+        )}
+        {dispensableReagents.map((reagent, reagentIndex) => (
+          <Button
+            key={reagentIndex}
+            className="chem-dispenser__dispense-buttons"
+            position="relative"
+            backgroundColor={""}
```

Comment:
Could just be `=""`, but do we need to set at all? Setting this to empty string specifically feels weird. May be implementation quirk of `Button`.